controllers/operands/kubevirt.go

```diff
@@ -280,41 +280,62 @@ func NewKubeVirt(hc *hcov1beta1.HyperConverged, opts ...string) (*kubevirtcorev1
 	return kv, nil
 }
 
-func hcoTuning2Kv(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.ReloadableComponentConfiguration, error) {
+func getHcoAnnotationTuning(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.ReloadableComponentConfiguration, error) {
 	var err error
-	if hc.Spec.TuningPolicy == hcov1beta1.HyperConvergedAnnotationTuningPolicy {
-		if annotation, ok := hc.Annotations[common.TuningPolicyAnnotationName]; ok {
-
-			var rates rateLimits
-			err := json.Unmarshal([]byte(annotation), &rates)
-			if err != nil {
-				return nil, err
-			}
+	if annotation, ok := hc.Annotations[common.TuningPolicyAnnotationName]; ok {
 
-			if rates.Qps <= 0 {
-				return nil, fmt.Errorf("qps parameter not found in annotation")
-			}
-			if rates.Burst <= 0 {
-				return nil, fmt.Errorf("burst parameter not found in annotation")
-			}
+		var rates rateLimits
+		err := json.Unmarshal([]byte(annotation), &rates)
+		if err != nil {
+			return nil, err
+		}
 
-			return &kubevirtcorev1.ReloadableComponentConfiguration{
-				RestClient: &kubevirtcorev1.RESTClientConfiguration{
-					RateLimiter: &kubevirtcorev1.RateLimiter{
-						TokenBucketRateLimiter: &kubevirtcorev1.TokenBucketRateLimiter{
-							QPS:   rates.Qps,
-							Burst: rates.Burst,
-						},
+		if rates.Qps <= 0 {
+			return nil, fmt.Errorf("qps parameter not found in annotation")
+		}
+		if rates.Burst <= 0 {
+			return nil, fmt.Errorf("burst parameter not found in annotation")
+		}
+		return &kubevirtcorev1.ReloadableComponentConfiguration{
+			RestClient: &kubevirtcorev1.RESTClientConfiguration{
+				RateLimiter: &kubevirtcorev1.RateLimiter{
+					TokenBucketRateLimiter: &kubevirtcorev1.TokenBucketRateLimiter{
+						QPS:   rates.Qps,
+						Burst: rates.Burst,
 					},
 				},
-			}, nil
-		} else {
-			err = fmt.Errorf("tuning policy set but annotation not present or wrong")
-		}
+			},
+		}, nil
+
+	} else {
+		err = fmt.Errorf("tuning policy set but annotation not present or wrong")
 	}
+
 	return nil, err
```

Comment:
I other branches in this function, we just return the error.

I think it will be the same logic, but a bit less confusing, to remove the `else` and just return the error. In this case, we don't need the `err` variable.